### PR TITLE
Get record labels

### PR DESCRIPTION
**Pull request by DanSava**
Enable storage to retrieve labels for labeled record data

### CODE REVIEW

File: src/ert_storage/endpoints/records.py [hunk 1]
```diff
@@ -462,12 +463,16 @@ async def get_ensemble_record(
...
     accept: str = Header("application/json"),
     realization_index: Optional[int] = None,
+    label: Optional[str] = None,
 ) -> Any:
     """
     Get record with a given `name`. If `realization_index` is not set, look for
     the ensemble-wide record. If it is set, look first for one created by a
     forward-model for the given realization index and then the ensemble-wide
     record.
+    If label is provided it is assumed the record data if of the form {"a": 1, "b": 2}
```
Comment (mortalisk) [inline src/ert_storage/endpoints/records.py:473]: if = is ?

File: src/ert_storage/endpoints/records.py [hunk 2]
```diff
@@ -462,12 +463,16 @@ async def get_ensemble_record(
...
+    and will retrun only the data for the provided label (i.e. label = "a" -> return: [[1]])
```
Comment (mortalisk) [inline src/ert_storage/endpoints/records.py:474]: return

**Review by mortalisk — APPROVED** [review]
fix typos, and LGTM

File: src/ert_storage/endpoints/records.py [hunk 3]
```diff
@@ -555,70 +655,76 @@ def get_ensemble_responses(
...
+        data = pd.DataFrame([[c[lbl_idx]] for c in matrix_content])
+    else:
+        data = pd.DataFrame(matrix_content)
+
+    if labels is not None:
+        data.columns = labels[0] if label is None else [label]
+        if realization_index is None and record.realization_index is not None:
+            data.index = [record.realization_index]
+        else:
+            data.index = labels[1] if realization_index is None else [realization_index]
```
Comment (frode-aarstad) [inline src/ert_storage/endpoints/records.py:689]: The logic in this block is rather confusing. Is it possible to simplify or add a comment about whats going on?

Comment (DanSava) [reply src/ert_storage/endpoints/records.py:689]: I will try to simplify and also add a comment the behaviour here is quite complex

**Review by frode-aarstad — DISMISSED** [review]
LGTM

**Review by mortalisk — APPROVED** [review]

**Pull request merged**